File: problems/1300_sum_of_mutated_array_closest_to_target/solution.py
class Solution:
    def findBestValue(self, arr: list[int], target: int) -> int:
        arr.sort()
        n = len(arr)
        prefix = [0] * (n + 1)
        for i in range(n):
            prefix[i + 1] = prefix[i] + arr[i]
        ans = 0
        min_diff = float('inf')
        for value in range(max(arr) + 1):
            import bisect
            idx = bisect.bisect_right(arr, value)
            total = prefix[idx] + value * (n - idx)
            diff = abs(total - target)
            if diff < min_diff:
                min_diff = diff
                ans = value
        return ans

File: problems/1300_sum_of_mutated_array_closest_to_target/Solution.java
import java.util.*;

class Solution {
    public int findBestValue(int[] arr, int target) {
        Arrays.sort(arr);
        int n = arr.length;
        int[] prefix = new int[n + 1];
        for (int i = 0; i < n; i++) prefix[i+1] = prefix[i] + arr[i];
        int ans = 0, minDiff = Integer.MAX_VALUE;
        for (int v = 0; v <= arr[n-1]; v++) {
            int idx = upperBound(arr, v);
            int total = prefix[idx] + v * (n - idx);
            int diff = Math.abs(total - target);
            if (diff < minDiff) { minDiff = diff; ans = v; }
        }
        return ans;
    }
    int upperBound(int[] arr, int val) {
        int lo = 0, hi = arr.length;
        while (lo < hi) { int mid = (lo+hi)/2; if (arr[mid] <= val) lo = mid+1; else hi = mid; }
        return lo;
    }
}
